pager/src/main/java/dev/chrisbanes/accompanist/pager/Pager.kt

```diff
@@ -0,0 +1,243 @@
+ *
+ * @param state the state object to be used to control or observe the list's state.
+ * @param modifier the modifier to apply to this layout.
+ * @param offscreenLimit the number of pages that should be retained on either side of the
+ * current page. This value is required to be `1` or greater.
+ * @param content a block which describes the content. Inside this block you can reference
+ * [PagerScope.currentPage] and other properties in [PagerScope].
+ */
+@ExperimentalPagerApi
+@Composable
+fun Pager(
+    state: PagerState,
+    modifier: Modifier = Modifier,
+    @IntRange(from = 1) offscreenLimit: Int = 1,
+    content: @Composable PagerScope.(page: Int) -> Unit
+) {
+    require(offscreenLimit >= 1) { "offscreenLimit is required to be >= 1" }
+
+    val reverseScroll = LocalLayoutDirection.current == LayoutDirection.Rtl
+
+    val density = LocalDensity.current
+    val decay = remember(density) { splineBasedDecay<Float>(density) }
+
+    val semantics = Modifier.composed {
+        val coroutineScope = rememberCoroutineScope()
+        semantics {
+            if (state.pageCount > 0) {
+                horizontalScrollAxisRange = ScrollAxisRange(
+                    value = { (state.currentPage + state.currentPageOffset) * state.pageSize },
+                    maxValue = { state.lastPageIndex.toFloat() * state.pageSize },
+                    reverseScrolling = reverseScroll
+                )
+                // Hook up scroll actions to our state
+                scrollBy { x: Float, _ ->
+                    coroutineScope.launch {
+                        state.draggableState.drag { dragBy(x) }
+                    }
+                    true
+                }
+                // Treat this as a selectable group
+                selectableGroup()
+            }
+        }
+    }
+
+    val draggable = Modifier.draggable(
+        state = state.draggableState,
+        startDragImmediately = true,
+        onDragStarted = {
+            state.selectionState = PagerState.SelectionState.Dragging
+        },
+        onDragStopped = { velocity ->
+            launch { state.performFling(velocity, decay) }
+        },
+        orientation = Orientation.Horizontal,
+        reverseDirection = reverseScroll,
+    )
+
+    Layout(
+        modifier = modifier.then(semantics).then(draggable),
+        content = {
+            val firstPage = (state.currentPage - offscreenLimit).coerceAtLeast(0)
+            val lastPage = (state.currentPage + offscreenLimit).coerceAtMost(state.lastPageIndex)
+
+            if (DebugLog) {
+                Log.d(
+                    LogTag,
+                    "Content: firstPage:$firstPage, " +
+                        "current:${state.currentPage}, " +
+                        "lastPage:$lastPage"
+                )
+            }
+
+            for (page in firstPage..lastPage) {
+                val pageData = PageData(page)
+                key(pageData) {
```

Comment:
pageData is data class so this works, but I would instead just write `key(page)`. people will copy your implementation and can break the contract accidentally. it is important that the value you pass in key() has a stable hashCode implementation otherwise the combined key can change when the process recreated which will break the state restoration